tc/sgx/trusted_worker_manager/enclave/CMakeVariables.txt

```diff
@@ -0,0 +1,191 @@
+ENDIF()
+
+SET(SGX_SDK "$ENV{SGX_SDK}")
+if("${SGX_SDK} " STREQUAL " ")
+    message(FATAL_ERROR "SGX_SDK environment variable not defined!")
+endif()
+SET(SGX_SSL "$ENV{SGX_SSL}")
+if("${SGX_SSL} " STREQUAL " ")
+    SET(SGX_SSL "/opt/intel/sgxssl")
+    message(STATUS "Setting default SGX_SSL=${SGX_SSL}")
+endif()
+
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+# ADDITIONAL LIBRARIES
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+# Intel SGX edger is a utility that generates edl artifacts
+SET(SGX_EDGER "${SGX_SDK}/bin/x64/sgx_edger8r")
+SET(SGX_SIGN "${SGX_SDK}/bin/x64/sgx_sign")
+
+SET(SGX_INCLUDE "${SGX_SDK}/include")
+SET(SGX_SSL_INCLUDE "${SGX_SSL}/include")
+SET(SGX_ENCLAVE_INCLUDE "${SGX_SDK}/include"
+                        "${SGX_SDK}/include/tlibc"
+                        "${SGX_SDK}/include/libcxx")
+
+SET(SGX_LIBS_TRUSTED_NAMES sgx_tstdc sgx_tstdcxx sgx_tcrypto)
+SET(SGX_LIBS_UNTRUSTED_NAMES )
+
+SET(SGX_LIBS_UNTRUSTED_NAMES ${SGX_LIBS_UNTRUSTED_NAMES} pthread )
+
+IF(${SGX_USE_SIMULATOR})
+    ADD_DEFINITIONS(-D SGX_SIMULATOR)
+    SET(SGX_LIBS_TRUSTED_NAMES sgx_trts_sim sgx_tservice_sim ${SGX_LIBS_TRUSTED_NAMES})
+    SET(SGX_LIBS_UNTRUSTED_NAMES sgx_urts_sim sgx_uae_service_sim ${SGX_LIBS_UNTRUSTED_NAMES})
+    SET(TRTS_LIBRARY_NAME "sgx_trts_sim")
+    SET(SERVICE_LIBRARY_NAME "sgx_tservice_sim")
+ELSE()
+    SET(SGX_LIBS_TRUSTED_NAMES sgx_trts sgx_tservice ${SGX_LIBS_TRUSTED_NAMES})
+    SET(SGX_LIBS_UNTRUSTED_NAMES sgx_urts sgx_uae_service ${SGX_LIBS_UNTRUSTED_NAMES})
+    SET(TRTS_LIBRARY_NAME "sgx_trts")
+    SET(SERVICE_LIBRARY_NAME "sgx_tservice")
+ENDIF()
+
+FOREACH(lib ${SGX_LIBS_TRUSTED_NAMES} ${SGX_LIBS_UNTRUSTED_NAMES})
+    SET(${lib} ${lib})
+ENDFOREACH(lib)
+
+FOREACH(lib ${SGX_LIBS_TRUSTED_NAMES})
+    SET(SGX_LIBS_TRUSTED ${SGX_LIBS_TRUSTED} ${${lib}})
+ENDFOREACH(lib)
+
+FOREACH(lib ${SGX_LIBS_UNTRUSTED_NAMES})
+    SET(SGX_LIBS_UNTRUSTED ${SGX_LIBS_UNTRUSTED} ${${lib}})
+ENDFOREACH(lib)
+
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+FUNCTION(SGX_EDGE_TRUSTED EDL EDGE_FILES)
+    GET_FILENAME_COMPONENT(EDL_BASE_NAME ${EDL} NAME_WE)
+    GET_FILENAME_COMPONENT(EDL_DIR_NAME ${EDL} DIRECTORY)
+    INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
+
+    SET (EDGE_FILES_LIST "${CMAKE_CURRENT_BINARY_DIR}/${EDL_BASE_NAME}_t.h" "${CMAKE_CURRENT_BINARY_DIR}/${EDL_BASE_NAME}_t.c")
+    SET (${EDGE_FILES} ${EDGE_FILES_LIST} PARENT_SCOPE)
+    ADD_CUSTOM_COMMAND( OUTPUT ${EDGE_FILES_LIST}
+	    COMMAND "${SGX_EDGER}" --trusted ${EDL} --search-path ${SINGLETON_PATH}
+	    --search-path ${KME_PATH} --search-path ${WPE_PATH} --search-path ${SGX_INCLUDE}
+	    --search-path ${SGX_SSL_INCLUDE} --search-path ${EDL_DIR_NAME}
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+        DEPENDS ${EDL}
+    )
+ENDFUNCTION()
+
+FUNCTION(SGX_SIGN_ENCLAVE TARGET KEY_FILE CONFIG)
```

Comment:
Can we keep the function definition in CMakeLists.txt file so that it can be used by children CMakeLists.txt belonging to singleton/kme/wpe?

Reply:
The Child Cmakelist will be able to access these function.
Please Refer $TCF_HOME/common/cpp Cmake files .